src/main/java/technology/dice/dicefairlink/discovery/tags/awsapi/ResourceGroupApiTagDiscovery.java

```diff
@@ -0,0 +1,72 @@
+package technology.dice.dicefairlink.discovery.tags.awsapi;
+
+import software.amazon.awssdk.services.resourcegroupstaggingapi.ResourceGroupsTaggingApiClient;
+import software.amazon.awssdk.services.resourcegroupstaggingapi.model.GetResourcesRequest;
+import software.amazon.awssdk.services.resourcegroupstaggingapi.paginators.GetResourcesIterable;
+import technology.dice.dicefairlink.config.FairlinkConfiguration;
+import technology.dice.dicefairlink.discovery.tags.ExclusionTag;
+import technology.dice.dicefairlink.discovery.tags.TagFilter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class ResourceGroupApiTagDiscovery implements TagFilter {
+  private static final Logger LOGGER =
+      Logger.getLogger(ResourceGroupApiTagDiscovery.class.getName());
+  private static final String RDS_DB_INSTANCE_FILTER = "rds:db";
+
+  private final ResourceGroupsTaggingApiClient client;
+  private final Collection<String> typeFilter;
+
+  public ResourceGroupApiTagDiscovery(FairlinkConfiguration fairlinkConfiguration) {
+    this.client =
+        ResourceGroupsTaggingApiClient.builder()
+            .region(fairlinkConfiguration.getAuroraClusterRegion())
+            .credentialsProvider(fairlinkConfiguration.getAwsCredentialsProvider())
+            .build();
+    this.typeFilter = new ArrayList<>(1);
+    this.typeFilter.add(RDS_DB_INSTANCE_FILTER);
```

Comment:
is this supposed to be mutable?

Reply:
This project doesn't have a dependency on Guava (only on test scope). Indeed it would be more elegant to use `ImmutableList.of(RDS_DB_INSTANCE_FILTER)`